base/abstractarray.jl

```diff
@@ -898,7 +903,21 @@ function copymutable(a::AbstractArray)
     @_propagate_inbounds_meta
     copyto!(similar(a), a)
 end
-copymutable(itr) = collect(itr)
+copymutable(itr) = ismutable(itr) ? copy(itr) : collect(itr)
+
+"""
+    ismutable(T)
+
+Predicate on the mutability of a collection of type `T`, which defaults to `false`.
+If `ismutable(T)` is `true`, `copy(::T)` must be defined.
+The definition `ismutable(x) = ismutable(typeof(x))` is provided for convenience so
+that instances can be passed instead of types. However the form that accepts a type
+argument should be defined for new types.
+"""
+ismutable(::Type) = false
+ismutable(a) = ismutable(typeof(a))
```

Comment:
Unfortunately we also have `isimmutable`...

Reply:
Ah right, I guess a new name must be found... `ismutablecollection`?